docs: add cache directory information. (#1871)
golangci/golangci-lint: docs/src/docs/usage/configuration.mdx

```yaml
{ .GolangciYaml }
```

## Cache

> GolangCI-Lint stores his cache into the [default user cache directory](https://golang.org/pkg/os/#UserCacheDir).

Reviewer: ```suggestion
GolangCI-Lint stores its cache in the [default user cache directory](https://golang.org/pkg/os/#UserCacheDir).
```

Definitely not "his", and probably not "into" (but I'm not a native speaker too).

> You can override the default cache directory with the environment variable `GOLANGCI_LINT_CACHE`, the path must be an absolute path.

Reviewer: ```suggestion
You can override the default cache directory with the environment variable `GOLANGCI_LINT_CACHE`; the path must be absolute.
```

Just to avoid repetition.
